eth/gasprice/gasprice_test.go

```diff
@@ -467,3 +472,62 @@ func TestSuggestGasPricePreAP3(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// Regression test to ensure the last estimation of base fee is not used
+// for the block immediately following a fee configuration update.
+func TestSuggestGasPriceAfterFeeConfigUpdate(t *testing.T) {
+	config := Config{
+		Blocks:     20,
+		Percentile: 60,
+	}
+
+	// create a chain config with fee manager enabled at genesis with [addr] as the admin
+	chainConfig := *params.TestChainConfig
+	chainConfig.FeeManagerConfig = precompile.NewFeeManagerConfig(big.NewInt(0), []common.Address{addr})
+
+	// create a fee config with higher MinBaseFee and prepare it for inclusion in a tx
+	signer := types.LatestSigner(params.TestChainConfig)
+	highFeeConfig := chainConfig.FeeConfig
+	highFeeConfig.MinBaseFee = big.NewInt(28_000_000_000)
+	data, err := precompile.PackSetFeeConfig(highFeeConfig)
+	require.NoError(t, err)
+
+	// before issuing the block changing the fee into the chain, the fee estimation should
+	// follow the fee config in genesis.
+	backend := newTestBackend(t, &chainConfig, 0, func(i int, b *core.BlockGen) {})
+	oracle, err := NewOracle(backend, config)
+	require.NoError(t, err)
+	got, err := oracle.SuggestPrice(context.Background())
+	require.NoError(t, err)
+	assert.Equal(t, chainConfig.FeeConfig.MinBaseFee, got)
```

Comment:
Can we use require throughout the test and do `require.New(t)` at the top so that we don't include `t` as a parameter?